internal/controller/controllers/agent_controller.go

```diff
@@ -132,6 +132,18 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, agent *adiiov1alph
 		return ctrl.Result{}, errors.New("Host not found in cluster")
 	}
 
+	internalHost, err := r.Installer.GetHostInternal(ctx, string(*c.ID), agent.Name)
+	if err != nil {
+		return ctrl.Result{Requeue: true, RequeueAfter: defaultRequeueAfterOnError}, err
+	}
+
+	if internalHost.Approved != spec.Approved {
+		err = r.Installer.UpdateHostApprovedInternal(ctx, string(*c.ID), agent.Name, spec.Approved)
+		if err != nil {
+			return ctrl.Result{Requeue: true, RequeueAfter: defaultRequeueAfterOnError}, err
```

Comment:
if `RequeueAfter` is set there is no need to add also `Requeue: true`

Reply:
Done